src/gridfs/upload.ts

```diff
@@ -147,8 +153,23 @@ export class GridFSBucketWriteStream extends Writable {
    * The stream is considered constructed when the indexes are done being created
    */
   override _construct(callback: (error?: Error | null) => void): void {
-    if (this.bucket.s.checkedIndexes) {
-      return process.nextTick(callback);
+    if (!this.bucket.s.calledOpenUploadStream) {
+      this.bucket.s.calledOpenUploadStream = true;
+
+      checkIndexes(this).then(
+        () => {
+          this.bucket.s.checkedIndexes = true;
+          this.bucket.emit('index');
+        },
+        error => {
+          if (error instanceof MongoOperationTimeoutError) {
+            handleError(this, error, callback);
+          } else {
+            squashError(error);
+            process.nextTick(callback);
```

Comment:
We're already inside of a promise rejection here, the nextTick made (alleged) sense when it was calling the callback sync but we're async here so we can just call the callback